### PR TITLE
Add Presto gamma_cdf function

### PR DESCRIPTION
**Pull request by arundpanicker**
Implementation of gamma cdf

### CODE REVIEW

**Comment by arundpanicker** [comment]
Fuzzer test output

```
I0704 23:28:52.896354 64061 ExpressionVerifier.cpp:57] ===================
I0704 23:28:52.896378 64061 ExpressionVerifier.cpp:69] All results match.
I0704 23:28:52.896452 64061 ExpressionFuzzer.cpp:1357] ==============================> Done with iteration 25
I0704 23:28:52.896889 64061 ExpressionFuzzer.cpp:1092] ==============================> Top 1 by number of rows processed
I0704 23:28:52.896898 64061 ExpressionFuzzer.cpp:1094] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I0704 23:28:52.896903 64061 ExpressionFuzzer.cpp:1098] gamma_cdf 2446 100.00% 141
I0704 23:28:52.897400 64061 ExpressionFuzzer.cpp:1104] ==============================> Bottom 1 by number of rows processed
I0704 23:28:52.897426 64061 ExpressionFuzzer.cpp:1106] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I0704 23:28:52.897529 64061 ExpressionFuzzer.cpp:1111] gamma_cdf 2446 100.00% 141
I0704 23:28:52.897567 64061 ExpressionFuzzer.cpp:1124] ==============================> All stats sorted by number of times the function was chosen
I0704 23:28:52.897578 64061 ExpressionFuzzer.cpp:1126] Format: functionName numTimesSelected proportionOfTimesSelected numProcessedRows
I0704 23:28:52.897589 64061 ExpressionFuzzer.cpp:1130] gamma_cdf 2446 100.00% 141
[==========] Running 0 tests from 0 test suites.
[==========] 0 tests from 0 test suites ran. (0 ms total)
[  PASSED  ] 0 tests.
```

**Comment by aditi-pandit** [comment]
@arundpanicker : Please can you check all the build failures.

Please also change the title to an active sentence : "Add Presto gamma_cdf function"

**Comment by amitkdutta** [comment]
@arundpanicker Please do a re-base. Otherwise looks good.

**Comment by arundpanicker** [comment]
Thanks @aditi-pandit , @amitkdutta for the review.
Rebase done.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp [hunk 1]
```diff
@@ -221,5 +221,30 @@ TEST_F(ProbabilityTest, invBetaCDF) {
...
 }
 
+TEST_F(ProbabilityTest, gammaCDF) {
+  const auto gammaCDF = [&](std::optional<double> shape,
+                            std::optional<double> scale,
+                            std::optional<double> value) {
+    return evaluateOnce<double>("gamma_cdf(c0, c1, c2)", shape, scale, value);
+  };
+
+  EXPECT_DOUBLE_EQ(0.96675918913720599, gammaCDF(0.5, 3.0, 6.8).value());
```
Comment (aditi-pandit) [inline velox/functions/prestosql/tests/ProbabilityTest.cpp:231]: Please add tests for values std::nullopt, kNan, kDoubleMax, kDoubleMin along with kInf

**Review by aditi-pandit — COMMENTED** [review]
Thanks. The rest of the PR looks good.

File: velox/functions/prestosql/tests/ProbabilityTest.cpp (hunk 1 above)
Comment (arundpanicker) [reply velox/functions/prestosql/tests/ProbabilityTest.cpp:231]: Thanks @aditi-pandit .
Added tests for values std::nullopt, kNan, kDoubleMax, kDoubleMin.

**Comment by yzhang1991** [comment]
Hi @arundpanicker, is this ready to review?

**Comment by arundpanicker** [comment]
Hi @yzhang1991 , this is ready for review.

**Comment by yzhang1991** [comment]
Thanks @arundpanicker. Please help ping the reviewer in the future to remind them.

@aditi-pandit please review. Thanks!

File: velox/docs/functions/presto/math.rst [hunk 2]
```diff
@@ -277,3 +281,4 @@ Probability Functions: inverse_cdf
     probability (p): P(N < n). The a, b parameters must be positive real values (all of type DOUBLE).
     The probability p must lie on the interval [0, 1].
 
+
```
Comment (aditi-pandit) [inline velox/docs/functions/presto/math.rst:284]: Please remove this newline.

File: velox/docs/functions/presto/math.rst [hunk 3]
```diff
@@ -266,6 +266,10 @@ Probability Functions: cdf
     The mean and value must be real values and the standard deviation must be a real and
     positive value (all of type DOUBLE).
 
+.. function:: gamma_cdf(shape, scale, value) -> double
```
Comment (aditi-pandit) [inline velox/docs/functions/presto/math.rst:269]: Please move this function to the right alphabetical order in the list.

File: velox/docs/functions/presto/math.rst (hunk 2 above)
Comment (arundpanicker) [reply velox/docs/functions/presto/math.rst]: Removed the new line.

File: velox/docs/functions/presto/math.rst (hunk 3 above)
Comment (arundpanicker) [reply velox/docs/functions/presto/math.rst]: Thanks @aditi-pandit . I have rearranged the function in the alphabetical order.

**Review by aditi-pandit — APPROVED** [review]
Thanks.

**Comment by arundpanicker** [comment]
Hi @Yuhta, requesting your review for this PR. Thanks.

**Comment by Yuhta** [comment]
@arundpanicker Can you rebase the code?

**Comment by arundpanicker** [comment]
Thanks @Yuhta . Rebase done

**Pull request closed without merging**